api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -593,6 +588,23 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 		Context.getPersonService().savePersonMergeLog(personMergeLog);
 	}
 	
+	private void requireNoActiveOrderOfSameType(Patient patient1, Patient patient2) {
+		String messageKey = "Patient.merge.cannotHaveSameTypeActiveOrders";
+		List<Order> ordersByPatient1 = Context.getOrderService().getAllOrdersByPatient(patient1);
+		List<Order> ordersByPatient2 = Context.getOrderService().getAllOrdersByPatient(patient2);
+		ordersByPatient1.forEach((Order order1) -> {
+			ordersByPatient2.forEach((Order order2) -> {
+				if (order1.isActive() && order2.isActive() && order1.getOrderType().equals(order2.getOrderType())) {
```

Comment:
Well, based on the discussion it was a suggestion that was never confirmed, i have posted to the talk topic to get a green light from others. Once we get the green light, it would be nice to clean up the tests methods i.e rename them to be more intuitive to portray what they are testing more clearly and also to have more useful assertions so don't end passing simply because the test data is not well set up. E.g if a test says that it should fail if both patients have active drug orders, you need to have an assertion for each patient that ensures that they actually  have at least one drug order before proceeding with the unit test.